templates/nagios_plugin.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2021 Canonical Ltd.
+
+# Authors:
+#   Robert Gildein <robert.gildein@canonical.com>
+
+import argparse
+import subprocess
+from configparser import ConfigParser
+
+import yaml
+from nagios_plugin3 import try_check, UnknownError, WarnError
+
+OPENSTACK_CREDENTIAL = {}
+
+
+def _load_openstack_credential(path):
+    """Load OpenStack credential cnf file."""
+    global OPENSTACK_CREDENTIAL
+    config = ConfigParser()
+    config.optionxform = str  # disables case conversion
+    config.read(path)
+
+    OPENSTACK_CREDENTIAL.update(dict(config.items("openstack")))
+
+
+def _get_credential_path():
+    """Get credential path from arguments."""
+    parser = argparse.ArgumentParser("check_openstack_networks")
+    parser.add_argument("--credential", "-c", required=True,
+                        help="Path to OpenStack credential cnf file")
+
+    args = parser.parse_args()
+    return args.credential
+
+
+def init_openstack():
+    """Initialize OpenStack credentials."""
+    credential_cnf_path = _get_credential_path()
+    _load_openstack_credential(credential_cnf_path)
+
+
+def openstack(*args):
+    """Run OpenStack command with credentials.
+
+    :returns: dictionary parsed from yaml output
+    :rtype: dict
+    """
+    output = subprocess.run(["openstack", *args, "--format=yaml"],
```

Comment:
The script should use `python3-openstackclient` rather than use the CLI. I suggest having a look to `charm-openstack-service-checks` for `check_nova_services.py` or other nrpe checks that use python libs.